src/sync/installation.ts

```diff
@@ -376,16 +377,19 @@ export const handleBackfillError = async (err,
 	}
 
 	logger.error({ err }, "Task failed, continuing with next task");
-
-	await markCurrentRepositoryAsFailedAndContinue(subscription, nextTask, scheduleNextTask);
+	await markCurrentRepositoryAsFailedAndContinue(subscription, nextTask, task, scheduleNextTask);
 };
 
-export const markCurrentRepositoryAsFailedAndContinue = async (subscription: Subscription, nextTask: Task, scheduleNextTask: (delayMs: number) => void): Promise<void> => {
+export const markCurrentRepositoryAsFailedAndContinue = async (subscription: Subscription, nextTask: Task, task: string,  scheduleNextTask: (delayMs: number) => void): Promise<void> => {
 	// marking the current task as failed
 	await updateRepo(subscription, nextTask.repositoryId, { [getStatusKey(nextTask.task)]: "failed" });
 	const gitHubProduct = getCloudOrServerFromGitHubAppId(subscription.gitHubAppId);
 	statsd.increment(metricTaskStatus.failed, [`type:${nextTask.task}`, `gitHubProduct:${gitHubProduct}`]);
 
+	if (task === "repository" && nextTask.task === "repository") {
```

Comment:
On around line 212 it has `const { task, cursor, repository } = nextTask;`, so isn't the `task` always === `nextTask.task`?